### PR TITLE
[flutter_tools] update pubspec .yaml message should not be displayed when creating a new plugin

### LINKED ISSUE #62594: "flutter create --platforms=android -t plugin" for a new project shows message "The `pubspec.yaml` under the project directory must be updated to support android"
`flutter create --platforms=android -t plugin` correctly creates the `android` directory.

### Steps to reproduce
```
$ flutter create --platforms=android -t plugin test_ada_plugin_android_only
Creating project test_ada_plugin_android_only...
  test_ada_plugin_android_only/LICENSE (created)
  test_ada_plugin_android_only/test/test_ada_plugin_android_only_test.dart (created)
  test_ada_plugin_android_only/test_ada_plugin_android_only.iml (created)
  test_ada_plugin_android_only/.gitignore (created)
  test_ada_plugin_android_only/.metadata (created)
  test_ada_plugin_android_only/android/settings.gradle (created)
  test_ada_plugin_android_only/android/gradle/wrapper/gradle-wrapper.properties (created)
  test_ada_plugin_android_only/android/gradle.properties (created)
  test_ada_plugin_android_only/android/.gitignore (created)
  test_ada_plugin_android_only/android/src/main/AndroidManifest.xml (created)
  test_ada_plugin_android_only/android/build.gradle (created)
  test_ada_plugin_android_only/android/test_ada_plugin_android_only_android.iml (created)
  test_ada_plugin_android_only/android/src/main/kotlin/com/example/test_ada_plugin_android_only/TestAdaPluginAndroidOnlyPlugin.kt (created)
  test_ada_plugin_android_only/pubspec.yaml (created)
  test_ada_plugin_android_only/README.md (created)
  test_ada_plugin_android_only/lib/test_ada_plugin_android_only.dart (created)
  test_ada_plugin_android_only/.idea/runConfigurations/example_lib_main_dart.xml (created)
  test_ada_plugin_android_only/.idea/libraries/Dart_SDK.xml (created)
  test_ada_plugin_android_only/.idea/modules.xml (created)
  test_ada_plugin_android_only/.idea/workspace.xml (created)
  test_ada_plugin_android_only/CHANGELOG.md (created)
Running "flutter pub get" in test_ada_plugin_android_only...         0.8s
The `pubspec.yaml` under the project directory must be updated to support android,
Add below lines to under the `platform:` key:


android:
 pluginClass: TestAdaPluginAndroidOnlyPlugin
 package: com.example.test_ada_plugin_android_only



If the `platforms` key does not exist in the `pubspec.yaml`, it might because that the plugin project does not
use the multi-platforms plugin format. We highly recommend a migration to the multi-platforms plugin format.
For detailed instructions on how to format the pubspec.yaml to support platforms using the multi-platforms format, see:
https://flutter.dev/docs/development/packages-and-plugins/developing-packages#plugin-platforms
...
```
The pubspec shows the Android platform:
```yml
flutter:
  # This section identifies this Flutter project as a plugin project.
  # The 'pluginClass' and Android 'package' identifiers should not ordinarily
  # be modified. They are used by the tooling to maintain consistency when
  # adding or updating assets for this project.
  plugin:
    platforms:
      android:
        package: com.example.test_ada_plugin_android_only
        pluginClass: TestAdaPluginAndroidOnlyPlugin
```

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

Fix a bug where the update `pubspec.yaml` message always display even when creating a new plugin.

## Related Issues

fixes https://github.com/flutter/flutter/issues/62594

## Tests

flutter create -t plugin in an empty folder should not show pubspec.yaml updating suggestion
flutter create -t plugin in an existing plugin should show pubspec.yaml updating suggestion

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/permeable/create_test.dart [hunk 1]
```diff
@@ -36,6 +37,7 @@ final Generator _kNoColorTerminalPlatform = () => FakePlatform.fromPlatform(cons
 final Map<Type, Generator> noColorTerminalOverride = <Type, Generator>{
   Platform: _kNoColorTerminalPlatform,
 };
+final BufferLogger logger = BufferLogger.test();
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/permeable/create_test.dart:40]: Add this in `main()` under `LoggingProcessManager loggingProcessManager;

**Review by jmagman — APPROVED** [review]
LGTM with nit

File: packages/flutter_tools/test/commands.shard/permeable/create_test.dart (hunk 1 above)
Comment (cyanglaz) [reply packages/flutter_tools/test/commands.shard/permeable/create_test.dart:40]: done

File: packages/flutter_tools/test/commands.shard/permeable/create_test.dart [hunk 2]
```diff
@@ -49,6 +49,7 @@ void main() {
   Directory projectDir;
   FlutterVersion mockFlutterVersion;
   LoggingProcessManager loggingProcessManager;
+  BufferLogger logger = BufferLogger.test();
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/permeable/create_test.dart:52]: Sorry, I also meant put the `logger = BufferLogger.test()` in setUp, match the pattern of `LoggingProcessManager`.

Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/permeable/create_test.dart:52]:
```suggestion
  BufferLogger logger;
```

Comment (cyanglaz) [reply packages/flutter_tools/test/commands.shard/permeable/create_test.dart]: done

Comment (cyanglaz) [reply packages/flutter_tools/test/commands.shard/permeable/create_test.dart]: Yeah I did that and forgot the remove the other constructor call. Fixed in a new commit.

**Review by jmagman — APPROVED** [review]
LGTM, thanks!

**Pull request merged**